### ISSUE TITLE
Emit a diagnostic if a `@type_check_only` symbol is imported in a runtime context

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#1559]
E.g.

````md

`stub.pyi`:

```pyi
from typing import type_check_only, Protocol

@type_check_only
class SupportsNext(Protocol):
    def __next__(self): ...

X: SupportsNext
```

`main.py`:

```py
# We should emit a diagnostic here:
# `SupportsNext` doesn't exist at runtime, it's only for the type checker
from stub import SupportsNext
```
````

The diagnostic should not fire if the `import` statement is inside a stub file. Ideally we would also suppress the diagnostic if the `import` statement is inside an `if TYPE_CHECKING` block, but doing that well depends on implementing https://github.com/astral-sh/ty/issues/1553.

**Comment by carljm** [comment]
It looks to me like neither pyright (not even in strict mode) nor mypy offer this diagnostic? Marking it as post-stable for that reason.
